### PR TITLE
feat: YAML data source

### PR DESCRIPTION
**Pull request by traut**
Adds `yaml` data source, similar to `json` source

In addition, some small documentation tweaks

### CODE REVIEW

File: internal/builtin/data_yaml.go [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+		},
+		Doc: `
+		Loads YAML files with the names that match provided ` + "`glob`" + ` pattern or a single file from provided ` + "`path`" + `value.
+
+		Either ` + "`glob`" + ` or ` + "`path`" + ` argument must be set.
+
+		When ` + "`path`" + ` argument is specified, the data source returns only the content of a file.
+		When ` + "`glob`" + ` argument is specified, the data source returns a list of dicts that contain the content of a file and file's metadata. For example:
+
+		` + "```yaml" + `
```
Comment (dobarx) [inline internal/builtin/data_yaml.go:51]: This looks like json, no?

File: internal/builtin/data_yaml.go [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+			]
+		  }
+		]
+		` + "```",
+	}
+}
+
+func fetchYAMLData(ctx context.Context, params *plugin.RetrieveDataParams) (plugindata.Data, diagnostics.Diag) {
+	glob := params.Args.GetAttrVal("glob")
+	path := params.Args.GetAttrVal("path")
```
Comment (dobarx) [inline internal/builtin/data_yaml.go:74]: Maybe we should throw an error when both attributes are set?

**Review by dobarx — APPROVED** [review]
Looks good! :+1: 
Just a couple of questions on small things.

File: internal/builtin/data_yaml.go (hunk 1 above)
Comment (traut) [reply internal/builtin/data_yaml.go:51]: ah, indeed!

**Pull request merged**